iroh-relay/src/protos/streams.rs

```diff
@@ -0,0 +1,168 @@
+                None => return Poll::Ready(None),
+                Some(Err(e)) => return Poll::Ready(Some(Err(e))),
+                Some(Ok(msg)) => {
+                    if msg.is_close() {
+                        // Indicate the stream is done when we receive a close message.
+                        // Note: We don't have to poll the stream to completion for it to close gracefully.
+                        return Poll::Ready(None);
+                    }
+                    if msg.is_ping() || msg.is_pong() {
+                        continue; // Responding appropriately to these is done inside of tokio_websockets/browser impls
+                    }
+                    if !msg.is_binary() {
+                        tracing::warn!(
+                            ?msg,
+                            "Got websocket message of unsupported type, skipping."
+                        );
+                        continue;
+                    }
+                    return Poll::Ready(Some(Ok(msg.into_payload().into())));
+                }
+            }
+        }
+    }
+}
+
+#[cfg(wasm_browser)]
+impl Stream for WsBytesFramed {
+    type Item = Result<Bytes, StreamError>;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        loop {
+            match ready!(Pin::new(&mut self.io).poll_next(cx)) {
+                None => return Poll::Ready(None),
+                Some(ws_stream_wasm::WsMessage::Binary(msg)) => {
+                    return Poll::Ready(Some(Ok(msg.into())))
+                }
+                Some(msg) => {
+                    tracing::warn!(?msg, "Got websocket message of unsupported type, skipping.");
+                    continue;
+                }
+            }
+        }
+    }
+}
+
+#[cfg(not(wasm_browser))]
+impl<T: AsyncRead + AsyncWrite + Unpin> Sink<Bytes> for WsBytesFramed<T> {
+    type Error = StreamError;
+
+    fn start_send(mut self: Pin<&mut Self>, bytes: Bytes) -> Result<(), Self::Error> {
+        let msg = tokio_websockets::Message::binary(tokio_websockets::Payload::from(bytes));
+        Pin::new(&mut self.io).start_send(msg)
+    }
+
+    fn poll_ready(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        Pin::new(&mut self.io).poll_ready(cx).map_err(Into::into)
+    }
+
+    fn poll_flush(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        Pin::new(&mut self.io).poll_flush(cx).map_err(Into::into)
+    }
+
+    fn poll_close(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        Pin::new(&mut self.io).poll_close(cx).map_err(Into::into)
+    }
+}
+
+#[cfg(wasm_browser)]
+impl Sink<Bytes> for WsBytesFramed {
+    type Error = StreamError;
+
+    fn start_send(mut self: Pin<&mut Self>, bytes: Bytes) -> Result<(), Self::Error> {
+        let msg = ws_stream_wasm::WsMessage::Binary(bytes.to_vec());
```

Comment:
https://docs.rs/bytes/latest/bytes/struct.Bytes.html#impl-From%3CBytes%3E-for-Vec%3Cu8%3E is more efficient